### PR TITLE
Cart wishlist

### PR DESCRIPTION
**Pull request by AnuragSahu11**
Features Added - 
*Added , add to cart, add to wishlist , remove from cart , remove from wishlist
*Created a separate context for cart and wishlist.
https://deploy-preview-5--proto-lucid-bookstore.netlify.app/productListing
(You can navigate through pages using navbar, cart icon , wishlist icon, and shop now link)

   Files to check  -
 - Files in /cart-wishlist/...
 - /context/cart-reducer.jsx
 - /context/cart-wishlist-context.jsx
 - /utility/check-in-list.jsx

### CODE REVIEW

File: lucid-book-store/src/components/cart-wishlist/cart-operations.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { useCart } from "../../context/cart-wishlist-context";
+import { checkInList } from "../../utility/check-in-list";
+
+const AddToCart = ({ product }) => {
+  const { cartState, cartDispatch } = useCart();
+  const clickHandler = () => {
+    if (checkInList(cartState.cart, product._id)) {
```
Comment (bajajcodes) [inline lucid-book-store/src/components/cart-wishlist/cart-operations.jsx:7]: I am guessing this line of code increases quantity if click again on add_to_cart button, if it is true then it's a good feature :)

File: lucid-book-store/src/components/cart-wishlist/cart-summary.jsx [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
-          <p className="m-y-0">$500</p>
-        </div>
-        <div className="subtitle width-100 flex-row regular space-between">
-          <p className="m-y-0">Discount</p>
-          <p className="m-y-0">-$100</p>
-        </div>
+        {cartState.cart.map((item) => (
+          <div className="subtitle m-up-1 width-100 flex-row regular space-between">
+            <p className="m-y-0">{item.title}({item.quantity} x)</p>
+            <p className="m-y-0">${item.price * item.quantity}</p>
```
Comment (bajajcodes) [inline lucid-book-store/src/components/cart-wishlist/cart-summary.jsx:14]: Computation can be separated as state.

File: lucid-book-store/src/components/cart-wishlist/cart-summary.jsx [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
-          <p className="m-y-0">-$50</p>
+          <p className="m-y-0">$50</p>
         </div>
         <hr />
         <div className="subtitle width-100 flex-row space-between is-3 m-y-1 is-dark">
-          <p className="m-y-0">Total</p>
-          <p className="m-y-0">-$2350</p>
+          <p className="m-y-0">Total {cartState.cart.length} items</p>
+          <p className="m-y-0">
+            {cartState.cart.reduce(
```
Comment (bajajcodes) [inline lucid-book-store/src/components/cart-wishlist/cart-summary.jsx:26]: Similarly computation can be separated from UI.

File: lucid-book-store/src/components/cart-wishlist/cart-card.jsx [hunk 4]
```diff
@@ -9,31 +16,30 @@ const CartCard = () => {
         </span>
         <span className="card-badge">Top seller</span>
```
Comment (bajajcodes) [inline lucid-book-store/src/components/cart-wishlist/cart-card.jsx:17]: This badge can also be a part of state.

**Review by Ronak-Mutha — COMMENTED** [review]
Good code structure and practices followed. Just do a spellcheck once and do changes if required. You can check initailCartValue once, for starter. Rest code looks good to merge.

**Review by bajajcodes — COMMENTED** [review]
- Code consistency is maintained from variable names to file names.
- Good PR review description, especially what files to check.
- Breaking of code into separate files and functions, makes code readable.

File: lucid-book-store/src/components/cart-wishlist/cart-operations.jsx (hunk 1 above)
Comment (AnuragSahu11) [reply lucid-book-store/src/components/cart-wishlist/cart-operations.jsx:7]: Yes if the product is in the cart then clicking on add to cart will just increase it's quantity

File: lucid-book-store/src/components/cart-wishlist/cart-summary.jsx (hunk 2 above)
Comment (AnuragSahu11) [reply lucid-book-store/src/components/cart-wishlist/cart-summary.jsx:14]: Yes I will make the changes

File: lucid-book-store/src/components/cart-wishlist/cart-summary.jsx (hunk 3 above)
Comment (AnuragSahu11) [reply lucid-book-store/src/components/cart-wishlist/cart-summary.jsx:26]: Yup

File: lucid-book-store/src/components/cart-wishlist/cart-card.jsx (hunk 4 above)
Comment (AnuragSahu11) [reply lucid-book-store/src/components/cart-wishlist/cart-card.jsx:17]: Yes

**Pull request merged**